components/worktemplates/components/preview.tsx

```diff
@@ -58,6 +71,10 @@ const Preview = ({template, className}: PreviewProps) => {
         };
     });
 
+    useEffect(() => {
+        dispatch(fetchListing());
+    }, [dispatch]);
```

Comment:
I'm not sure if it's a standard or good practice to observe dispatch? In theory, we want to retrieve this only once on the first load.

```suggestion
    }, []);
```

Reply:
I am used to adding dispatch in the dependency list as usually the lint complains about it but for our application it doesn't so I will remove it.